api/inference/v1alpha1/config_types.go

```diff
@@ -42,7 +42,6 @@ type BackendRuntimeConfig struct {
 	// ConfigName represents the recommended configuration name for the backend,
 	// It will be inferred from the models in the runtime if not specified, e.g. default,
 	// speculative-decoding.
-	// +kubebuilder:default=default
```

Comment:
> Why remove the default value of the ConfigName field?

llmaz [infers the recommended configuration name](https://github.com/InftyAI/llmaz/blob/v0.1.3/pkg/controller_helper/helper.go#L36) for the backend if not specified. However, the `kubebuilder:default=default` annotation prevents this inference by always setting ConfigName to "default" instead of leaving it as nil, bypassing the role-based detection logic by mistake.

Reply:
Set the default value here doesn't make any difference, right? The inference is just a guardrail I believe.

Reply:
If we set the default value here, when we don't define `configName` in the Playground, even though we define main and draft models, the configName will always be set as `default` instead of `speculative-decoding`.

Reply:
Recalled why we remove this before. Make sense to me.